src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7915,53 +7963,43 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                     LookupOptions.Default,
                     originalBinder: this,
                     diagnose: false,
-                    ref discardedUseSiteInfo);
+                    ref useSiteInfo);
 
                 if (lookupResult.IsMultiViable)
                 {
                     foreach (var candidate in lookupResult.Symbols)
                     {
                         if (!candidate.IsStatic &&
                             candidate is PropertySymbol property &&
-                            IsAccessible(property, ref discardedUseSiteInfo) &&
+                            IsAccessible(property, ref useSiteInfo) &&
                             property.OriginalDefinition is { ParameterCount: 1 } original &&
-                            isIntNotByRef(original.Parameters[0]))
+                            original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
                         {
-                            CheckImplicitThisCopyInReadOnlyMember(receiverOpt, lengthOrCountProperty.GetMethod, diagnostics);
                             // note: implicit copy check on the indexer accessor happens in CheckPropertyValueKind
-                            patternIndexerAccess = new BoundIndexOrRangePatternIndexerAccess(
-                                syntax,
-                                receiverOpt,
-                                lengthOrCountProperty,
-                                property,
-                                BindToNaturalType(argument, diagnostics),
-                                property.Type);
-                            break;
+                            patternSymbol = property;
+                            checkWellKnown(WellKnownMember.System_Index__GetOffset);
+                            return true;
                         }
                     }
                 }
             }
             else if (receiverType.SpecialType == SpecialType.System_String)
```

Comment:
>receiverType.SpecialType == SpecialType.System_String [](http://example.com/codeflow?start=21&length=53)

If we have a string, we should be unconditionally returning true. Missing `Substring` is a problem that needs to be reported to the user of the malformed Corlib.